azkaban-common/src/test/java/azkaban/executor/container/KubernetesContainerizedImplTest.java

```diff
@@ -262,7 +262,7 @@ public void testPodConstruction() throws Exception {
     allImageTypes.addAll(jobTypes);
     allImageTypes.addAll(dependencyTypes);
     final VersionSet versionSet = this.kubernetesContainerizedImpl
-        .fetchVersionSet(flow.getExecutionId(), flowParam, allImageTypes, flow);
+        .fetchVersionSet(flow.getExecutionId(), new Props(null, flowParam), allImageTypes, flow);
```

Comment:
Can you please add valid test case which will test project props and precedence as well?

Reply:
I created this PR for reference only. Will work on production code once we agree on the approach.